src/gei/Commands/MigrateRepoCommand.cs

```diff
@@ -196,6 +267,83 @@ public async Task Invoke(string githubSourceOrg, string adoSourceOrg, string ado
+          string ghesApiUrl,
+          string githubSourceOrg,
+          string sourceRepo,
+          string azureStorageConnectionString,
+          bool noSslVerify = false
+          )
+        {
+            _log.LogInformation($"GHES API URL: {ghesApiUrl}");
+
+            if (string.IsNullOrWhiteSpace(azureStorageConnectionString))
+            {
+                _log.LogInformation("--azure-storage-connection-string not set, using environment variable AZURE_STORAGE_CONNECTION_STRING");
+                azureStorageConnectionString = _environmentVariableProvider.AzureStorageConnectionString();
+
+                if (string.IsNullOrWhiteSpace(azureStorageConnectionString))
+                {
+                    throw new OctoshiftCliException("Please set either --azure-storage-connection-string or AZURE_STORAGE_CONNECTION_STRING");
+                }
+            }
+
+            if (noSslVerify)
+            {
+                _log.LogInformation("SSL verification disabled");
+            }
+
+            var ghesApi = noSslVerify ? _sourceGithubApiFactory.CreateClientNoSsl(ghesApiUrl) : _sourceGithubApiFactory.Create(ghesApiUrl);
+            var azureApi = noSslVerify ? _azureApiFactory.CreateClientNoSsl(azureStorageConnectionString) : _azureApiFactory.Create(azureStorageConnectionString);
+
+            var gitDataArchiveId = await ghesApi.StartGitArchiveGeneration(githubSourceOrg, sourceRepo);
+            _log.LogInformation($"Archive generation of git data started with id: {gitDataArchiveId}");
+            var metadataArchiveId = await ghesApi.StartMetadataArchiveGeneration(githubSourceOrg, sourceRepo);
+            _log.LogInformation($"Archive generation of metadata started with id: {metadataArchiveId}");
+
+            var gitArchiveUrl = await WaitForArchiveGeneration(ghesApi, githubSourceOrg, gitDataArchiveId);
+            _log.LogInformation($"Archive (git) download url: {gitArchiveUrl}");
+            var metadataArchiveUrl = await WaitForArchiveGeneration(ghesApi, githubSourceOrg, metadataArchiveId);
+            _log.LogInformation($"Archive (metadata) download url: {metadataArchiveUrl}");
+
+            var timeNow = $"{DateTime.Now:yyyy-MM-dd_HH-mm-ss}";
+
+            var gitArchiveFileName = $"{timeNow}-{gitDataArchiveId}-{GIT_ARCHIVE_FILE_NAME}";
+            var metadataArchiveFileName = $"{timeNow}-{metadataArchiveId}-{METADATA_ARCHIVE_FILE_NAME}";
+
+            _log.LogInformation($"Downloading archive from {gitArchiveUrl}");
+            var gitArchiveContent = await azureApi.DownloadArchive(gitArchiveUrl);
+            _log.LogInformation($"Downloading archive from {metadataArchiveUrl}");
+            var metadataArchiveContent = await azureApi.DownloadArchive(metadataArchiveUrl);
+
+            _log.LogInformation($"Uploading archive {gitArchiveFileName} to Azure Blob Storage");
+            var authenticatedGitArchiveUrl = await azureApi.UploadToBlob(gitArchiveFileName, gitArchiveContent);
```

Comment:
nit: Since it's returning a Uri it's better to name the variable accordingly:
```suggestion
            var authenticatedGitArchiveUri = await azureApi.UploadToBlob(gitArchiveFileName, gitArchiveContent);
```